src/Tempest/Database/src/ConnectionInitializer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use Tempest\Container\Container;
+use Tempest\Container\Initializer;
+use Tempest\Container\Singleton;
+use Tempest\Core\AppConfig;
+
+final class ConnectionInitializer implements Initializer
+{
+    private static Connection|null $instance = null;
```

Comment:
Doesn't `#[Singleton]` solve this?

Reply:
Gotcha. It's unfortunate that we need to include test-specific code into the source though. Maybe we could register a separate initializer instead of this one while running tests? Something like `TestConnectionInitializer`?

Reply:
That seems like an incredible good idea. Lets do that.

Reply:
Implemented it using the `CachedConnectionInitializer` and adding the initializer in the `FrameworkIntegrationTestCase` . It was incredably straight forward to do. Kudos to you all, amazingly designed.

Reply:
Aaah nice! 😁